scripts/convert.js

```diff
@@ -5,17 +5,19 @@
 const csv = require('csv');
 const fs = require('fs');
 const mapValues = require('lodash/mapValues');
+const kebabCase = require('lodash/kebabCase');
 const getStdin = require('get-stdin');
 const path = require('path');
 
-const dataShellPath = path.resolve(__dirname, '../resources/data-shell.json');
-const dataShell = JSON.parse(fs.readFileSync(dataShellPath, 'utf-8'));
+const dataShellPath = path.resolve(__dirname, '../src/constants/data-shell.json');
+const dataShell = require(dataShellPath);
 const dict = mapValues(dataShell.data.indices, (atts) => atts.source);
 
 getStdin().then(rawCsv => {
   csv.parse(rawCsv, { columns: true }, (err, data) => {
     process.stdout.write(JSON.stringify(data.map((row) => ({
-      name: row['Cities'],
+      name: row.Cities,
+      id: kebabCase(row.Cities),
       indices: mapValues(dict, (colKey) => {
         if (!row[colKey]) throw new Error(`Key not found! ${colKey}`);
         return row[colKey];
```

Comment:
I'm not convinced that the ability to pass in any input and output path is a good thing. How does a user of this script know where to output the file?

I'm a fan of developing for today, and right now the front end code relies on the data being output to a specific directory, but this code doesn't enforce outputting the data to that directory. So this is flexible, but if you specified something other than the `DATA_URL` constant, it fail silently since the new data would be ignored and the old data would still be used.

I think this should either use the `DATA_URL` URL in this file, or if we want to allow for it to have a different output at runtime, define a script in `package.json` that defines where the output must go:

`"generate-city-data": "node scripts/convert.js < resources/cities.csv > public/data/smartCitiesData.json"`

Since that documents the expected input/output too. Also we can then call this from the `build` script to ensure we always have the most recent data.

If we wind up having a different script to go to S3 or something, we update the script.

Reply:
I like the idea of putting specifics in package.json. However, these suggested changes aren't in scope for this PR.